.eslintrc.js

```diff
@@ -41,5 +41,44 @@ module.exports = {
         'prefer-object-spread/prefer-object-spread': 'off',
       },
     },
+
+    // files that are not allowed to use devDepenedncies
+    {
+      files: ['packages/kbn-ui-framework/**/*'],
+      excludedFiles: [
+        'packages/kbn-ui-framework/**/*.test.js',
+        'packages/kbn-ui-framework/doc_site/**/*',
+        'packages/kbn-ui-framework/generator-kui/**/*',
+        'packages/kbn-ui-framework/Gruntfile.js',
+      ],
+      rules: {
+        'import/no-extraneous-dependencies': [
+          'error',
+          {
+            devDependencies: false,
+            peerDependencies: true,
+          },
+        ],
+      },
+    },
+
+    // files that are allowed to use devDepenedncies
+    {
+      files: [
+        'packages/kbn-ui-framework/**/*.test.js',
+        'packages/kbn-ui-framework/doc_site/**/*',
+        'packages/kbn-ui-framework/generator-kui/**/*',
+        'packages/kbn-ui-framework/Gruntfile.js',
+      ],
+      rules: {
+        'import/no-extraneous-dependencies': [
+          'error',
+          {
+            devDependencies: true,
```

Comment:
Why do we need this setup for `ui-framework` but not for the rest of Kibana? Maybe it's been discussed somewhere (I've been sick the last week, so my brain is still a bit foggy).

Or is it just a "nice addon" to help make it clear?

Reply:
I only added it to the ui-framework to help me verify that I was pulling in the right dependencies and making the right things dev/normal deps. I think we could absolutely use this in other places, but I don't think it needs to go into this PR.